zebra-network/src/peer/client.rs

```diff
@@ -253,31 +261,53 @@ impl<T: std::fmt::Debug> Drop for MustUseOneshotSender<T> {
 impl Client {
     /// Check if this connection's heartbeat task has exited.
     fn check_heartbeat(&mut self, cx: &mut Context<'_>) -> Result<(), SharedPeerError> {
-        if let Poll::Ready(()) = self
+        let is_canceled = self
```

Comment:
Note: this change ignores the `JoinHandle` return type, using `matches!(_, Poll::Ready(()))` would check it at compile time.

That's probably ok, because the return type is unlikely to change.

Reply:
Same as above.